stash_engine/lib/tasks/stash_engine_tasks.rake

```diff
@@ -136,6 +136,12 @@ namespace :identifiers do
         end
       end
       StashEngine::InternalDatum.create(identifier_id: datum.identifier_id, data_type: 'publicationName', value: title) unless title.blank?
+      # Submit the info to Solr if published/embargoed
+      identifier = StashEngine::Identifier.where(id: datum.identifier_id).first
+      if identifier.present? && identifier.latest_resource.present?
+        current_ca = identifier.latest_resource.current_curation_activity
+        identifier.latest_resource.submit_to_solr if current_ca.present? && (current_ca.published? || current_ca.embargoed?)
```

Comment:
I'm not sure if it matters, but you may want to use something like `identifier.latest_resource_with_public_metadata` instead, since I think `latest_resource` might include ones that haven't been set to embargoed or published yet or that are still being edited by the user.

I might also be doing this wrong when I write metadata, but I think maybe there it just uses the resource that had its state changed for submission.

Reply:
oh good catch. that would be more efficient for sure